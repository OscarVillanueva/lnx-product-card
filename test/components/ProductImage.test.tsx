import React from 'react'
import renderer from 'react-test-renderer'
import { ProductCard } from '../../src/components';
import products from '../data/products'

describe('ProductTitle Text', () => {
  it('renders without crashing - with custom title', () => {
    
    const wrapper = renderer.create(
      <ProductCard.Image 
        image='https://resource.logitechg.com/w_1000,c_limit,q_auto,f_auto,dpr_auto/d_transparent.gif/content/dam/gaming/en/products/g915-tkl/g915-tkl-gallery-2-carbon-new.png?v=1'
      />
    )

    expect(wrapper.toJSON()).toMatchSnapshot()

  });

  test('with product name', () => { 

    const wrapper = renderer.create(
      <ProductCard
        product = { products[1] }
      >
        { () => (
          <ProductCard.Image />
        )}
      </ProductCard>
    )

    expect(wrapper.toJSON()).toMatchSnapshot()

  })
});
