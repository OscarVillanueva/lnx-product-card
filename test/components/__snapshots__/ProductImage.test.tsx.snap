// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle Text renders without crashing - with custom title 1`] = `
<img
  alt="A preview of the product"
  className="productImg undefined"
/>
`;

exports[`ProductTitle Text with product name 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="A preview of the product"
    className="productImg undefined"
    src="./coffee-mug2.png"
  />
</div>
`;
